src/main/java/uk/gov/companieshouse/api/testdata/service/impl/BarcodeServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.companieshouse.api.testdata.service.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.companieshouse.api.testdata.exception.BarcodeServiceException;
+import uk.gov.companieshouse.api.testdata.service.BarcodeService;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+
+import static uk.gov.companieshouse.api.testdata.constants.ErrorMessageConstants.BARCODE_ERROR;
+
+@Service
+public class BarcodeServiceImpl implements BarcodeService {
+
+    private RestTemplate template;
+
+    @Value("${barcode.url}")
+    private String barcodeUrl;
+
+    public void setTemplate(RestTemplate template) {
```

Comment:
Should we reduce the visibility of this method? I think we only have it for unit testing so `protected` or (default) should be fine.